src/redux/books/books.js

```diff
@@ -20,16 +28,77 @@ export default function books(state = defaultState, action) {
   }
 }
 
-export function addBook(payload) {
+function loadBooks(payload) {
+  return {
+    type: LOAD_BOOKS,
+    payload,
+  };
+}
+
+function addBook(payload) {
   return {
     type: ADD_BOOK,
     payload,
   };
 }
 
-export function removeBook(payload) {
+function removeBook(payload) {
   return {
     type: REMOVE_BOOK,
     payload,
   };
 }
+
+export function addBookAPI(payload) {
+  const bookDetails = { ...payload, item_id: Date.now() };
+  return async function addBookThunk(dispatch) {
+    fetch(
+      `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${APP_ID}/books`,
+      {
+        method: 'POST',
+        body: JSON.stringify(bookDetails),
+        headers: {
+          'content-type': 'application/json',
+        },
+      },
+    ).then(
+      (response) => response.status === 201 && dispatch(addBook(bookDetails)),
+    );
+  };
+}
+
+export function loadBooksAPI() {
+  function arrayFormat(respObj) {
+    const values = Object.values(respObj).map((item) => item[0]);
+    return Object.keys(respObj).map((item, i) => ({
+      id: Number(item),
+      ...values[i],
+    }));
+  }
+
+  return async function loadBooksThunk(dispatch) {
+    fetch(
+      `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${APP_ID}/books`,
+    )
+      .then((response) => response.json())
+      .then((json) => dispatch(loadBooks(arrayFormat(json))));
+  };
+}
```

Comment:
- [ ] Let's make a good JS by using ES6 arrow functions. and async and await too